woocommerce/woocommerce
# Product Collection: Inspector control to change selected product

### Changes proposed in this Pull Request:





Closes #44877.

This is a follow up PR to #50164. In this PR, I have implemented Inspector control to change selected product. This way, merchant can change the selected product using Inspector control. Here is how it looks like:
![image](https://github.com/user-attachments/assets/aeb4e762-5c22-4e9f-8df9-053793a3c93f)

Once user click on selected product button, it will show a Popover to search and select a different product. As soon as a new product will be selected, it will reflect in the inspector control:
![image](https://github.com/user-attachments/assets/d326cfc8-589a-45f0-81fb-4612bc562fb9)



### How to test the changes in this Pull Request:

**Pre-requisite:** Install this plugin - 
[register-product-collection-tester.zip](https://github.com/user-attachments/files/16362323/register-product-collection-tester.zip)



**Case 1: Adding "My Custom Collection - Product Context" to a place where product context doesn't exist**

1. Create a new post and add Product Collection block.
2. Choose "My Custom Collection - Product Context" block.
3. Choose any product from Editor product picker.
4. Verify that the selected product is shown in Inspector control.
![image](https://github.com/user-attachments/assets/d6e75174-6f8d-4549-9438-ac58bc531680)
5. Click on selected product and verify that a Popover is shown to search and select a different product.
6. Search and select a different product and verify that it reflects in Inspector control.
7. _To be verified by developers only_: 
  - In the code editor, verify that `productReference` attribute is updated with new product id of selected product. When a product is changed, `productReference` attribute should be updated with new product id.

**Case 2: Adding "My Custom Collection - Product Context" on Single Product template**

1. Go to Single Product template and add Product Collection block.
2. Choose "My Custom Collection - Product Context" block.
3. Verify that product picker is not shown in Inspector control.

**Case 3: Adding "My Custom Collection - Product Context" on Specific Product template**

1. Create a new Specific Product template (See screenshot below)
![image](https://github.com/user-attachments/assets/7cbfe19e-e97e-4ca7-988c-32a8ee1a3fd4)

2. Go to specific product template that you created in step 1 and add Product Collection block.
3. Verify that product picker is not shown in Inspector control.

**Case 4: Collection which doesn't require product context, shouldn't show product picker**

1. Create a new post and add Product Collection block.
2. Choose any other collection except "My Custom Collection - Product Context".
3. Verify that product picker is not shown in Inspector control.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Product Collection - Implement Inspector control to change selected product
</details>

When a collection require a product context, but if it doesn't exist in current template/page/post etc. then we will show a control in inspector controls to change the selected product. This way, merchant can manually provide a product context to the collection.

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
